lib/src/rules/prefer_collection_literals.dart

```diff
@@ -40,23 +43,47 @@ class PreferCollectionLiterals extends LintRule implements NodeLintRule {
       [LinterContext context]) {
     final visitor = new _Visitor(this);
     registry.addInstanceCreationExpression(this, visitor);
+    registry.addMethodInvocation(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
   final LintRule rule;
-
   _Visitor(this.rule);
 
   @override
-  void visitInstanceCreationExpression(InstanceCreationExpression node) {
-    if (node.constructorName.name == null &&
-        node.argumentList.arguments.isEmpty &&
-        (DartTypeUtilities.isClass(node.staticType, 'List', 'dart.core') ||
-            DartTypeUtilities.isClass(node.staticType, 'Map', 'dart.core') ||
-            DartTypeUtilities.isClass(
-                node.staticType, 'LinkedHashMap', 'dart.collection'))) {
+  void visitMethodInvocation(MethodInvocation node) {
+    // ['foo', 'bar', 'baz'].toSet();
+    if (node.methodName.name != 'toSet') {
+      return;
+    }
+    if (node.target is ListLiteral) {
       rule.reportLint(node);
     }
   }
+
+  @override
+  void visitInstanceCreationExpression(InstanceCreationExpression node) {
+    final constructorName = node.constructorName.name?.name;
+
+    // Lists, Maps.
+    if (DartTypeUtilities.isClass(node.staticType, 'List', 'dart.core') ||
+        DartTypeUtilities.isClass(node.staticType, 'Map', 'dart.core') ||
+        DartTypeUtilities.isClass(
+            node.staticType, 'LinkedHashMap', 'dart.collection')) {
+      if (constructorName == null && node.argumentList.arguments.isEmpty) {
+        rule.reportLint(node);
+      }
+      return;
+    }
+
+    // Sets.
+    if (DartTypeUtilities.isClass(node.staticType, 'Set', 'dart.core') ||
+        DartTypeUtilities.isClass(
+            node.staticType, 'LinkedHashSet', 'dart.collection')) {
+      if (constructorName != 'identity' && constructorName != 'unmodifiable') {
```

Comment:
It would be safer to make this a whitelist rather than a blacklist. As unlikely as it seems, it means that we won't start linting new `Set` constructors accidentally.

Reply:
Agreed.  Fixed.